src/main/kotlin/io/snyk/plugin/SnykBulkFileListener.kt

```diff
@@ -123,43 +145,53 @@ class SnykBulkFileListener : BulkFileListener {
     ) {
-        val iacCache = toolWindowPanel.currentIacResult
-        val iacFiles = iacCache?.allCliIssues ?: return
-        val project = toolWindowPanel.project
-
-        runBackgroundableTask("Updating Snyk Infrastructure As Code Cache...", project, true) {
-            DumbService.getInstance(project).runReadActionInSmartMode {
-                var changed = false
-
-                val newIacFileList = iacFiles.toMutableList()
-                virtualFilesAffected
-                    .filter { iacFileExtensions.contains(it.extension) }
-                    .filter { ProjectRootManager.getInstance(project).fileIndex.isInContent(it) }
-                    .forEach {
-                        changed = true // for new files we need to "dirty" the cache, too
-                        newIacFileList.forEachIndexed { i, iacIssuesForFile ->
-                            if (pathsEqual(it.path, iacIssuesForFile.targetFilePath)) {
-                                val obsoleteIacFile = makeObsolete(iacIssuesForFile)
-                                newIacFileList[i] = obsoleteIacFile
-                            }
-                        }
-                    }
+        if (virtualFilesAffected.isEmpty()) return
+        val toolWindowPanel = getSnykToolWindowPanel(project)
+        val iacFiles = toolWindowPanel?.currentIacResult?.allCliIssues
+        if (iacFiles == null || iacFiles.isEmpty()) return
+
+        var changed = false
 
-                if (changed) {
-                    val newIacCache = IacResult(newIacFileList.toImmutableList(), null)
-                    toolWindowPanel.currentIacResult = newIacCache
-                    toolWindowPanel.displayIacResults(newIacCache)
-                    toolWindowPanel.iacScanNeeded = true
-                    DaemonCodeAnalyzer.getInstance(toolWindowPanel.project).restart()
+        val newIacFileList = iacFiles.toMutableList()
+        virtualFilesAffected
+            .filter { iacFileExtensions.contains(it.extension) }
+            .filter { ProjectRootManager.getInstance(project).fileIndex.isInContent(it) }
+            .forEach {
+                changed = true // for new files we need to "dirty" the cache, too
+                newIacFileList.forEachIndexed { i, iacIssuesForFile ->
+                    if (pathsEqual(it.path, iacIssuesForFile.targetFilePath)) {
+                        val obsoleteIacFile = makeObsolete(iacIssuesForFile)
+                        newIacFileList[i] = obsoleteIacFile
+                    }
                 }
             }
+
+        if (changed) {
+            val newIacCache = IacResult(newIacFileList.toImmutableList(), null)
+            toolWindowPanel.currentIacResult = newIacCache
+            toolWindowPanel.iacScanNeeded = true
+            ApplicationManager.getApplication().invokeLater {
+                toolWindowPanel.displayIacResults(newIacCache)
+            }
+            DaemonCodeAnalyzer.getInstance(toolWindowPanel.project).restart()
```

Comment:
Why is this outside the invokeLater?

Reply:
displayIacResults require EDT while inside SnykBulkFileListener we seems to be in some background thread.